Gem/Code/Source/Components/NetworkMatchComponent.cpp

```diff
@@ -193,6 +200,21 @@ namespace MultiplayerSample
         }
     }
 
+    void NetworkMatchComponentController::OnPlayerArmorZero(Multiplayer::NetEntityId playerEntity)
+    {
+        const auto playerIterator = AZStd::find(m_players.begin(), m_players.end(), playerEntity);
+        const auto playerHandle = Multiplayer::GetNetworkEntityManager()->GetEntity(playerEntity);
```

Comment:
You can move this inside if () to avoid the look up if the player is not in the list.